File: Q32 Valid palindrome II.js

    var function palin(s, i, j) {
        while (j >= i) {
            if (s[i] !== s[j]) {
                return false;
            } else {
                i++;
                j--;
            }
        }
        return true;
    }

    var function validPalindrome(s) {
        let i = 0;
        let j = s.length - 1;
        while (j >= i) {
            if (s[i] !== s[j]) {
                return this.palin(s, i + 1, j) || this.palin(s, i, j - 1);
            } else {
                i++;
                j--;
            }
        }
        return true;
    }

File: Q32 Valid palindrome II .py
class Solution:
    def palin(self, s, i, j):
        while j >= i:
            if s[i] != s[j]:
                return False
            else:
                i += 1
                j -= 1
        return True

    def validPalindrome(self, s: str) -> bool:
        i = 0
        j = len(s) - 1
        while j >= i:
            if s[i] != s[j]:
                return self.palin(s, i + 1, j) or self.palin(s, i, j - 1)
            else:
                i += 1
                j -= 1
        return True

